cmd/tempo-query/tempo/plugin.go

```diff
@@ -316,19 +348,45 @@ func (b *Backend) FindTraces(req *storage_v1.FindTracesRequest, stream storage_v
 	}
 
 	span.AddEvent(fmt.Sprintf("Found %d trace IDs", len(resp.TraceIDs)))
+	b.logger.Info("FindTraces: fetching traces", zap.Int("traceids", len(resp.TraceIDs)))
+
+	numWorkers := b.findTracesConcurrentRequests
+	jobs := make(chan job, len(resp.TraceIDs))
+	results := make(chan jobResult, len(resp.TraceIDs))
+	var wg sync.WaitGroup
+	// Start workers
+	for w := 0; w < numWorkers; w++ {
+		go worker(b, jobs, results, &wg)
```

Comment:
instead of passing `&wg` here, we could do `go func() { defer wg.Done(); worker(...) }`

Reply:
The `wg` is used to track number of jobs (find trace requests, `wg.Add(len(resp.TraceIDs))`) so it needs to be executed inside the worked.